depmap-compute/depmap_compute/context.py

```diff
@@ -101,3 +112,30 @@ def walk(node, key):
         return node
 
     return walk(expr, None)
+
+
+def decode_slice_id(slice_id) -> tuple[str, str, str]:
+    """
+    Originally based on the function of the same name from vector_catalog.SliceSerializer,
+    Data Explorer 2 slice ids are a superset of the legacy slice ids.
+    Originally, slice IDs were formatted like "slice/some_dataset_id/some_feature_label/label", 
+    or "slice/some_dataset_id/some_feature_id/entity_id", where the last part of the string
+    (originally called the SliceRowType) specifies whether the feature is being identified by ID or by label. 
+
+    The DE2 slice IDs give you flexibility by letting you query samples as well using the "transpose_label" specifier.
+    When "transpose_label" is used as the last segment of the slice ID, it means we should query for samples.
+    """
+    parts = slice_id.split("/")
+    assert (
+        parts[0] == "slice" and len(parts) >= 4 and len(parts) <= 5
+    ), f"Malformed slice_id: {slice_id}"
+
+    if len(parts) == 5:
+        # handle dataset IDs with slashes in them
+        parts[1:3] = ["/".join(parts[1:3])]
+
+    dataset_id = unquote(parts[1])
+    dimension_identifier = unquote(parts[2])
+    slice_type = unquote(parts[3])  # "label", "entity_id", or "transpose_label"
```

Comment:
The original implementation used an enum for this:
https://github.com/broadinstitute/depmap-portal/blob/53bee77ccceedde16cd2efcde57581034bbca3ac/portal-backend/depmap/vector_catalog/models.py#L13-L15

I didn't do the same because I didn't want the hack to seem "official." But like we were saying above, DE2 has been out in the wild long enough that it's de facto official now.

Reply:
Thanks, I agree it would be nice to have this defined by an enum now that it's more official. 

I wasn't sure if there are other slice identifier types we want to support going forward (for example, do we want to be able to look up both features and samples by ID? or neither?) - so I added a note to the breadbox discussion list. If we all agree about that, I'll define this as an enum and add some better docs 👍 
https://app.asana.com/0/1206925945720127/1208125648857065/f